### PR TITLE
poke: 2.4 -> 3.0

### PR DESCRIPTION
**Pull request by trofi**
Changes: https://www.jemarch.net/poke-3.0-relnotes.html

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 212675` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>poke</li>
  </ul>
</details>

**Comment by trofi** [comment]
Fails build when `autoconf` triggers, possibly incomplete `gnulib` bindling:

```
configure:8964: error: possibly undefined macro: gl_ANYTHREADLIB_EARLY
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure:23758: error: possibly undefined macro: gl_PTHREADLIB
configure:23843: error: possibly undefined macro: gl_WEAK_SYMBOLS
autoreconf: error: /nix/store/pacd8pk0k2gfmsvs2qlysrgil09hgjns-autoconf-2.71/bin/autoconf failed with exit status: 1
```

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 212675` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>poke</li>
  </ul>
</details>

**Comment by trofi** [comment]
3 test failures:

```
...
FAIL: poke.pkl/in-1.pk execution test
FAIL: poke.pkl/in-2.pk execution test
FAIL: poke.pkl/in-3.pk execution test
Running /build/poke-3.0/testsuite/poke.pktest/pktest.exp ...
Running /build/poke-3.0/testsuite/poke.repl/repl.exp ...
Running /build/poke-3.0/testsuite/poke.std/std.exp ...
Running /build/poke-3.0/testsuite/poke.time/time.exp ...

                === poke Summary ===

# of expected passes            7419
# of unexpected failures        3
# of unsupported tests          3
```

Need more time to triage.

**Comment by trofi** [comment]
Rebased against `staging` as it depends on https://github.com/NixOS/nixpkgs/pull/212688. Ready for review.

**Review by AndersonTorres — DISMISSED** [review]

**Comment by wegank** [comment]
The build seems to fail on aarch64-darwin.

**Comment by trofi** [comment]
> The build seems to fail on aarch64-darwin.

Yeah, due to `staging` target it times out building `llvm`: `error: building of '/nix/store/q2cwsidw0a2041pgxp6ir9dxz2z8mfbn-llvm-11.1.0.drv!lib,out' from .drv file timed out after 3600 seconds`.

Switched to `draft` for now.

**Comment by wegank** [comment]
> > The build seems to fail on aarch64-darwin.
> 
> Yeah, due to `staging` target it times out building `llvm`: `error: building of '/nix/store/q2cwsidw0a2041pgxp6ir9dxz2z8mfbn-llvm-11.1.0.drv!lib,out' from .drv file timed out after 3600 seconds`.
> 
> Switched to `draft` for now.

Indeed, but I mean that it fails even when it is rebased on master...

**Comment by trofi** [comment]
> > > The build seems to fail on aarch64-darwin.
> > 
> > 
> > Yeah, due to `staging` target it times out building `llvm`: `error: building of '/nix/store/q2cwsidw0a2041pgxp6ir9dxz2z8mfbn-llvm-11.1.0.drv!lib,out' from .drv file timed out after 3600 seconds`.
> > Switched to `draft` for now.
> 
> Indeed, but I mean that it fails even when it is rebased on master...

With https://github.com/NixOS/nixpkgs/pull/212688 or without applied? It's expected to fail without in test phase. If it fails earlier then it's a bigger problem and we need to restore a form of `config.*` file mangling.

**Comment by wegank** [comment]
It fails earlier: [log.txt](https://github.com/NixOS/nixpkgs/files/10517013/log.txt)

**Comment by trofi** [comment]
Aha. I'll try to reproduce on `x86_64-darwin` in hopes that it fails the same. Thank you!

**Comment by wegank** [comment]
The build succeeds on x86_64-darwin, however...

File: pkgs/applications/editors/poke/default.nix [hunk 1]
```diff
@@ -14,6 +13,10 @@
 , textStylingSupport ? true
 , dejagnu
 
+# to regenerate ./configure
```
Comment (AndersonTorres) [inline pkgs/applications/editors/poke/default.nix:16]: `autoreconfHook` doesn't work?

Comment (trofi) [reply pkgs/applications/editors/poke/default.nix:16]: Yeah, unmodified `autoreconfHook` fails as https://github.com/NixOS/nixpkgs/pull/212675#issuecomment-1404293951

**Comment by trofi** [comment]
ofborg managed to build `aarch64-darwin`. Ready to review.

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 212675` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>poke</li>
  </ul>
</details>

**Pull request closed without merging**